keras/callbacks.py

```diff
@@ -567,19 +573,22 @@ def __init__(self, count_mode='samples',
     def on_train_begin(self, logs=None):
         self.verbose = self.params['verbose']
         self.epochs = self.params['epochs']
+        self.log_values = []
+        if self.target is 0:
```

Comment:
You can remove the warning, since it will not be informative for the user (it assumes too much context)